src/System.Memory/src/System/Buffers/Text/Utf8Formatter/FormattingHelpers.cs

```diff
@@ -189,6 +215,44 @@ public static int CountDigits(ulong value)
             return digits;
         }
 
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static int CountDigits(uint value)
+        {
+            int digits = 1;
+            if (value >= 1000)
```

Comment:
Is the 1000/1000000 split better than having just one 100000 split?

Reply:
I didn't play around with this too much. Did a 3-way split similar to ulong. I could try to splitting it in 2. What's the rationale for the split boundary here (i.e. why 100000 instead of the current 1000000)?

Reply:
Whether 3-way split or 2-way split is better depends on relative performance of multiplication and conditional jumps.